src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -48,6 +56,10 @@ public function __construct($name, NodeInterface $parent = null)
 
         $this->name = $name;
         $this->parent = $parent;
+
+        // Backwards compatibility for old property names, to be removed in 3.0
+        $this->normalizationClosures = &$this->normalizationCallbacks;
+        $this->finalValidationClosures = &$this->finalValidationCallbacks;
```

Comment:
Yes, using magic methods would allow allow deprecation notices and also somewhat simpler implementation (no need for `__clone()` tricks). It would also make the properties effectively public, which could be a problem. In addition the new properties could also probably be private, unless there's some specific reason the old ones were protected.

All of these changes look like bikeshedding to me though. I checked with Packanalyst earlier and found just one class that would need to be changed for 3.0 compatibility -- and that one uses the setter methods. A quick search is not going to find every use case, but I think using the properties directly is likely rare even in custom subclasses.

Of course, I can still make the changes if there's agreement that using `__get()` and `__set()` would be preferred.